packages/paste-core/components/tabs/src/Tabs.tsx

```diff
@@ -48,13 +48,32 @@ export interface TabsProps extends TabPrimitiveInitialState {
 const Tabs = React.forwardRef<HTMLDivElement, TabsProps>(
   ({ children, element, orientation = "horizontal", state, variant, ...initialState }, ref) => {
     // If returned state from primitive has orientation set to undefined, use the default "horizontal"
+    const [prevSelectedTab, setPrevSelectedTab] = React.useState<string | undefined>(undefined);
     const { orientation: tabOrientation = orientation, ...tab } =
       state || useTabPrimitiveState({ orientation, ...initialState });
     const elementName = getElementName(tabOrientation, "TABS", element);
     const value = React.useMemo(
       () => ({ ...tab, orientation: tabOrientation, variant }),
-      [...Object.values(tab), tabOrientation, variant],
+      [tab, tabOrientation, variant],
     );
+
+    const { selectedId } = tab;
+    // Scroll to the selected tab if it exists on mount
+    React.useEffect(() => {
+      if (typeof selectedId === "string") {
+        setTimeout(() => {
+          document
+            // eslint-disable-next-line unicorn/prefer-query-selector
+            .getElementById(selectedId)
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore - behavior is typed incorrectly in Typescript v4, fixed in v5+
+            ?.scrollIntoView({ behavior: prevSelectedTab === undefined ? "instant" : "smooth" });
```

Comment:
Hi @krisantrobus,

This PR introduced some weird behavior since it applies to all `Tabs` variations including vertical ones. When used with vertical tabs `scrollIntoView` obviously scrolls both vertically and horizontally until a tab header is visible which causes an annoying experience when you need to scroll up to revert this unintended view change.

Could this be please fixed?

Reply:
Hey @dmitriy-kudelko apologies for this. Working on a fix and will update once we have a PR up

Reply:
PR is up and we are reviewing internally: https://github.com/twilio-labs/paste/pull/4143

Reply:
Cool. Thanks!

Reply:
@krisantrobus , great news. Thank you for implementing this feature and for quickly releasing a fix! 👍 
We've been waiting for it and have already applied it to some places on the UI.

BTW, there seems to be also an issue with the inability to scroll tabs till the very end. Clicking on the right arrow doesn't seem to help either. The left boundary doesn't have this problem. Not sure if it's something on our end, but we don't have any margin/padding customizations for the Tabs component which could explain that. It doesn't seem to depend on the window width either. I attached a video:

https://github.com/user-attachments/assets/3f987474-c635-46a6-b5f0-526d8d826e3c

Reply:
Hmm this is very strange and something we have not encountered. Could you share more details. What is the viewport width you are using and number of tabs. If you could replicate in a codesandbox that would be be very helpful. Also are you able to use the tab key to highlight a tab and use arrow keyboard buttons to get to the very last item?

Reply:
@dmitriy-kudelko I could not replicate it in our strorybooks when making some page compositions, however I did find it in a deploy preview for our Paste upgrade in one of our internal tools. I will be working on this again tomorrow and we should see another path. Again, so sorry for these issues

Reply:
Hi @dmitriy-kudelko I have validated the fix in 20.18.1 and I can confirm the issue has been resolved.

Reply:
Hi @krisantrobus , nice! Thank you so much!